File: ros_ws/code/turtle_spawn_client.cpp
// Include the necessary ROS header file for C++ programming
#include <ros/ros.h>

// Include the header file for the Spawn Service
#include <turtlesim/Spawn.h>


int main(int argc, char** argv){
	// initialize a node named 'spawn_client'
	ros::init(argc,argv,"turtle_spawn_client");
	
	// create a nodeHandle object to interface with the ROS network.
	ros::NodeHandle nh;
	
	// wait until the the /spawn service become available 
	ros::service::waitForService("/spawn");
	
	// get the service handler
	ros::ServiceClient  new_turtle  =  nh.serviceClient < turtlesim::Spawn >("/spawn");

    // Initialize request data for the services to create a new turtle
    //  at (7.5,7.5) with a direction (pi/2) under the name "turtle2".
    turtlesim::Spawn  new_turtle_srv;
    new_turtle_srv.request.x  =  7.5;
    new_turtle_srv.request.y  =  7.5;
	new_turtle_srv.request.theta  =  1.57;
    new_turtle_srv.request.name  =  "turtle2";
	
	// Call the spawn service to  The mode will be blocked
    // waiting for a response from the server (return value)
    new_turtle.call(new_turtle_srv);

	// print the returned value
    std::cout << "Spwan turtle successfully [name:" << 
	          new_turtle_srv.response.name.c_str() << "]" << std::endl;

	return 0;
}

File: ros_ws/code/turtle_spawn_client.py
#!/usr/bin/env python3
# use the default python compiler

# import ROS python library
import rospy

#import the Spawn Service 
from  turtlesim.srv  import  Spawn

# initialize a node named 'spawn_client'
rospy.init_node ('spawn_client')

# wait until the the /spawn service become available 
rospy.wait_for_service('/spawn') 

try :
    # get the service handler
    new_turtle =  rospy.ServiceProxy('/spawn', Spawn) 
    
    # Call the spawn service to create a new turtle at (7.5,7.5) with a 
    # direction (pi/2) under the name "turtle2". The mode will be   
    # blocked waiting for a response from the server (return value)
    response = new_turtle(7.5, 7.5, 1.57, "turtle2") 
    
    #print the returned value
    print("The service responded by {}".format(response.name));
    
except  rospy.ServiceException as e :
    # report a failure
    print  ("failed to call service : %s" % e)
    
    
    